engine/src/test/java/org/camunda/bpm/engine/test/cockroachdb/CockroachDBRetriesTest.java

```diff
@@ -0,0 +1,505 @@
+    public Void execute(CommandContext commandContext) {
+
+      tries++;
+      throw new CrdbTransactionRetryException("The Process Engine Bootstrap has failed.");
+
+    }
+
+    public int getTries() {
+      return tries;
+    }
+
+    @Override
+    public boolean isRetryable() {
+      return super.isRetryable();
+    }
+  }
+
+  protected static class ControllableFailingCommand extends ControllableCommand<Void> {
+
+    protected int tries = 0;
+
+    @Override
+    public Void execute(CommandContext commandContext) {
+
+      monitor.sync();
+
+      tries++;
+      throw new CrdbTransactionRetryException("Does not retry");
+    }
+
+    public int getTries() {
+      return tries;
+    }
+
+    @Override
+    public boolean isRetryable() {
+      return false;
+    }
+  }
+
+  protected class ControllableJobDefinitionPriorityCommand extends ControllableCommand<Void> {
+
+    protected SetJobDefinitionPriorityCmd jobDefinitionPriorityCmd;
+
+    public ControllableJobDefinitionPriorityCommand(String jobDefinitionId, Long priority, boolean cascade) {
+      this.jobDefinitionPriorityCmd = new SetJobDefinitionPriorityCmd(jobDefinitionId, priority, cascade);
+    }
+
+    @Override
+    public Void execute(CommandContext commandContext) {
+
+      monitor.sync();
+
+      jobDefinitionPriorityCmd.execute(commandContext);
+
+      monitor.sync();
+
+      return null;
+    }
+  }
+
+  protected class ControllableJobSuspensionCommand extends ControllableCommand<Void> {
+
+    protected SuspendJobDefinitionCmd suspendJobDefinitionCmd;
+
+    public ControllableJobSuspensionCommand(String processDefinitionKey) {
+      UpdateJobDefinitionSuspensionStateBuilderImpl builder = new UpdateJobDefinitionSuspensionStateBuilderImpl()
+          .byProcessDefinitionKey(processDefinitionKey)
+          .includeJobs(true);
+
+      this.suspendJobDefinitionCmd = new SuspendJobDefinitionCmd(builder);
+    }
+
+    @Override
+    public Void execute(CommandContext commandContext) {
+
+      monitor.sync();
+
+      suspendJobDefinitionCmd.execute(commandContext);
+
+      monitor.sync();
+
+      return null;
+    }
+  }
+
+  protected void closeDownProcessEngine() {
+    final ProcessEngine otherProcessEngine = ProcessEngines.getProcessEngine(PROCESS_ENGINE_NAME);
+    if (otherProcessEngine != null) {
+
+      ((ProcessEngineConfigurationImpl)otherProcessEngine.getProcessEngineConfiguration())
+          .getCommandExecutorTxRequired()
+          .execute((Command<Void>) commandContext -> {
+
+            List<Job> jobs = otherProcessEngine.getManagementService().createJobQuery().list();
+            if (jobs.size() > 0) {
+              assertEquals(1, jobs.size());
+              String jobId = jobs.get(0).getId();
+              commandContext.getJobManager().deleteJob((JobEntity) jobs.get(0));
+              commandContext.getHistoricJobLogManager().deleteHistoricJobLogByJobId(jobId);
+            }
+
+            return null;
+          });
```

Comment:
Removed it. It's not actually necessary. The reason was to remove any data created by the additional Process Engine that is created by the `shouldRethrowBootstrapEngineOleWhenRetriesAreExhausted()` test case. However, since the test simulates a scenario where the Process  Engine bootstrap fails, and the transaction is rolled back, there is nothing to clean up.